### DISCUSSION TITLE
Route not being matched

### DISCUSSION THREAD

**Discussion by vsergione** [mockoon/mockoon#934] [Help]
Hello!

I'm having the following situation: I've mocked an API using Mockoon on my Linux PC. I've deployed the JSON config file on another Linux machine to run the mocked API on it.
 
For a while everything was working fine until recently when I can see that a request which previously was successfully matched and mocked by Mockoon, it is not matching anymore. What changed in the meantime is the client application. But if I check the logs and I inspect visually the requested URL, it doesn't make sense at all, why the route is not being matched. Is it possible for the URL to contain any "invisible" characters which are not actually visible in the log?
 
Any ideas??

**Comment by 255kb** [comment]
Never heard of such issue happening due to an extra character. 
What we did recently is to introduce the routes folders which required a data migration. What could have happened is that the new version of the app migrated the data file, but you are still trying to run it with an older CLI. But this is a wild guess as it shouldn't work. You should receive a "data too recent" error. 
Could you please share the data file you are trying to run, so I could try to reproduce the issue?

**Reply by vsergione** [reply]
I've been investigating the problem further and I found out that actually none of the mocked routes where served. The weird part was that when running on the local machine, either from the GUI or from the CLI, everything was running perfect.

But then I've realized that the mockoon-cli running in the container was the latest version (as per my docker-compose file), while the version running on my PC is older. So I've started running various versions of mockoon/cli from docker hub and found out that this issue affects the latest 2 tags:

https://hub.docker.com/layers/mockoon/cli/latest/images/sha256-76fc5ec225736093095af01f8a31655c8269bd0df09bd3fb8956963d22e31545?context=explore
and https://hub.docker.com/layers/mockoon/cli/2.3.0/images/sha256-76fc5ec225736093095af01f8a31655c8269bd0df09bd3fb8956963d22e31545?context=explore

**Reply by 255kb** [reply]
I just tested running a data file from the previous version (migration 24, before the folder system) and run it with the latest CLI (2.3.0) with and without Docker and the file was correctly migrated and the routes were working. 

So, it's working as intended on my side.

I may be able to investigate a bit more if you provide the data file you are using.

**Comment by vsergione** [comment]
Sure. Please find it attached.

**Reply by 255kb** [reply]
I don't see any attachment.

**Reply by vsergione** [reply]
Sorry. I've missed it. Hope it works now.
[meps.zip](https://github.com/mockoon/mockoon/files/10704476/meps.zip)

**Reply by 255kb** [reply]
I just tested in the same way, with the latest CLI and with the latest Docker image and it's working as intended. Data file gets migrated and I can call the routes you defined.
Could you show me an example of an URL you try to call?

**Reply by vsergione** [reply]
I will try. It will take a while, as we're getting this error only in k8s cluster.
